doc/users.md

```diff
@@ -1 +1,1712 @@
+```
+
+```
+{
+    "status_code": "cdc-404",
+    "status_message": "user id not found",
+    "data": null
+}
+```
+
+```
+{
+    "status_code": "CDC-200",
+    "status_message": "OK",
+    "data": {
+        "id":1,
+        "name"         : "dennis",
+        "username"     : "vdennis",  
+        "type_akses"   : 4,  
+        "company_id"   : "5fb5df66-5f80-416d-a055-401bcf944ef7",
+        "region_id"    : "5fb5e66d-e47c-47fd-b4bf-4064cf944ef7", 
+        "branch_id"    : "5fb5e437-69dc-4eb6-8f8c-4022cf944ef7",  
+        "kios_id"      : "5fb5e626-65ac-473d-af2c-3ff2cf944ef7", 
+        "created"      : "2020-12-02 09:03:44",
+        "modified"     : "2020-12-14 12:07:14"
+    }
+}
+```
+
+### Logic
+- is_login, last_login, last_activity are not required in stateless concepts
+
+type_akses list :
+- 0 = SUPER USER
+- 1 = COMPANY
+- 2 = REGION
+- 3 = AHASS
+- 4 = POS/KIOS
+- 5 = ALL-WMS
+- 6 = WMS-NETWORK DEV
+- 7 = WMS-SE
+- 8 = WMS-Salesman PMD
+
+for edit, you need get :
+- if type_akses given then company_id, branch_id, region_id, kios_id is required depend on the value.
+- company_id from companies tables if type_akses in range 1-4.
+- branch_id from branches tables if type_akses in range 3-4.
+- region_id from regions tables if type_akses is 2.
+- kios_id from kios tables if type_akses is 4.
+
+#### Validation
+- name: not empty
+- type_akses: must in range (1-8)
+- company_id: required and not empty and must be registered in companies if type_akses in range (1-4)
+- region_id: required and not empty and must be registered in regions if type_akses = 2
+- branch_id: required and not empty and must be registered in branches if type_akses in range (3-4)
+- kios_id: required and not empty and must be registered in kios if type_akses = 4
+
+### Scenario Test
+
+#### Case : Negative Case 1
+
+Request Payload : empty
+
+Response HTTP Status Code : 200
+
+Response Payload :
+```
+{
+    "status_code": "cdc-200",
+    "status_message": "no change is requested",
+    "data": null
+}
+```
+
+#### Case : Negative Case 2
+
+Request Payload :
+```
+{}
+```
+
+Response HTTP Status Code : 200
+
+Response Payload :
+```
+{
+    "status_code": "cdc-200",
+    "status_message": "no change is requested",
+    "data": null
+}
+```
+
+#### Case : Negative Case 3
+
+Request payload :
+```
+{
+    "name": ""
+}
+```
+
+Response HTTP Status Code : 400
+
+Response Payload :
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "name is empty",
+    "data": null
+}
+```
+
+#### Case : Negative Case 4
+
+Request Payload
+```
+{
+    "type_akses": 9
+}
+```
+
+Response HTTP Status Code : 400
+
+Response Payload
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "type_akses must in range 1-8",
+    "data": null
+}
+```
+
+#### Case : Negative Case 5
+
+Request Payload
+```
+{
+    "type_akses": 1
+}
+```
+
+Response HTTP Status Code : 400
+
+Response Payload
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "company_id is required",
```

Comment:
in edit mode, if there is no company_id in request payload, you can use the existing data. its mean, the company_id is not changed.